internal/updater/providers/slickvpn/location.go

```diff
@@ -0,0 +1,117 @@
+package slickvpn
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"regexp"
+
+	"github.com/qdm12/gluetun/internal/models"
+	htmlutils "github.com/qdm12/gluetun/internal/updater/utils/html"
+	"golang.org/x/net/html"
+)
+
+const (
+	ColContinent = 1
+	ColCountry   = 2
+	ColCity      = 3
+	ColConfig    = 4
+)
+
+var (
+	ErrExtractServerFromLink   = errors.New("unable to extract server string from link")
+	ErrNoMatchingServer        = errors.New("couldn't find matching server")
+	ErrOpenvpnLinkNodeNotFound = errors.New("couldn't find OpenVPN link node")
+)
+
+func fetchLocationInfo(
+	ctx context.Context,
+	client *http.Client,
+	indexURL string,
+) (serversOut []models.Server, err error) {
+	request, err := http.NewRequestWithContext(ctx, http.MethodGet, indexURL, nil)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create HTTP request: %w", err)
+	}
+
+	response, err := client.Do(request)
+	if err != nil {
+		return nil, fmt.Errorf("HTTP request failed: %w", err)
+	}
+	defer response.Body.Close()
+
+	doc, err := html.Parse(response.Body)
+	if err != nil {
+		return nil, fmt.Errorf("an error occurred parsing response body: %w", err)
+	}
+
+	rows := htmlutils.GetNodesBySelector(doc, "table#location-table > tbody > tr")
+
+	rowCount := 0
+	for _, rowNode := range rows {
+		rowCount++
+		colCount := 0
+		server := models.Server{}
+		for cellNode := rowNode.FirstChild; cellNode != nil; cellNode = cellNode.NextSibling {
+			colCount++
+			switch colCount {
+			case ColCountry:
+				// Country
+				server.Country = cellNode.FirstChild.Data
+			case ColCity:
+				// City
+				server.City = cellNode.FirstChild.Data
+			case ColConfig:
+				// OVPN configs
+				linkNodes := htmlutils.GetNodesBySelector(cellNode, "a[href$='.ovpn']")
+				for _, linkNode := range linkNodes {
+					if htmlutils.GetText(linkNode) == ovpnLinkText {
+						configLink, err := htmlutils.GetAttr(linkNode, "href")
+						if err != nil {
+							return nil, err
+						}
+
+						serverName, err := extractServerFromLink(configLink)
+						if err != nil {
+							return nil, err
+						}
+						server.Hostname = serverName
+						break
+					}
+				}
+			case ColContinent:
+				// Continent. We don't use this (yet?) - fallthrough to default (skip)
+				fallthrough
+			default:
+				continue // Skip
+			}
+		}
+		serversOut = append(serversOut, server)
+	}
+	return serversOut, nil
+}
+
+func extractServerFromLink(link string) (serverName string, err error) {
+	re := regexp.MustCompile(`^.+\/(?P<serverName>.+)\.ovpn$`)
```

Comment:
this is going to sound very strange, but that should be at global scope. In example:

```go
var serverNameRegex = regexp.MustCompile(`^.+\/(?P<serverName>.+)\.ovpn$`)
```

That way if the regex expression is wrong, it will panic at start and it's easy to detect. Better panic right at start than later at runtime. And we do want to use `MustCompile` (the panic flavor) since a bad regex would really just be a programming error so we want to panic the program (ideally detect at compilation time but oh well compilation in Go is fast for a reason I guess)

This also saves the regex compilation every time we use this function.